src/ucp/core/ucp_am.c

```diff
@@ -597,45 +696,75 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_data_recv_nbx,
 }
 
 static UCS_F_ALWAYS_INLINE ucs_status_t
-ucp_am_handler_common(ucp_worker_h worker, void *hdr_end, size_t hdr_size,
-                      size_t total_length, ucp_ep_h reply_ep, uint16_t am_id,
-                      unsigned am_flags)
-{
-    ucp_recv_desc_t *desc = NULL;
-    ucs_status_t status;
-    unsigned flags;
-    void *arg;
-
-    if (ucs_unlikely(!ucp_am_recv_check_id(worker, am_id))) {
+ucp_am_invoke_cb(ucp_worker_h worker, ucp_am_hdr_t *am_hdr,
+                 size_t am_hdr_length, size_t data_length,
+                 ucp_ep_h reply_ep, int has_desc)
+{
+    uint16_t        am_id = am_hdr->am_id;
```

Comment:
unusual alignment
below we don't use it

Reply:
it's an experiment;) looks nice imo

Reply:
> looks nice imo

yes, looks good. I'm ok